src/frontend/src/optimizer/mod.rs

```diff
@@ -909,7 +917,12 @@ impl PlanRoot {
             RequiredDist::ShardByKey(bitset)
         };
 
-        let stream_plan = inline_session_timezone_in_exprs(context, stream_plan)?;
+        let mut stream_plan = inline_session_timezone_in_exprs(context, stream_plan)?;
+
+        if !not_null_keys.is_empty() {
```

Comment:
Why does it called `keys`? IIUC the `NOT NULL` constraint can be applied to any columns. 🤔

Reply:
Updated. Thx!